src/triggers.ts

```diff
@@ -0,0 +1,85 @@
+const closestElement: (selector: string, target: Element) => Element | null = (
```

Comment:
Could we perhaps make this refactoring in a standalone PR? That way each change becomes easier to review.